src/nmea.h

```diff
@@ -0,0 +1,138 @@
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file nmea.h
+ *
+ * NMEA protocol definitions
+ *
+ * @author WeiPeng Guo <guoweipeng1990@sina.com>
+ * @author Stone White <stone@thone.io>
+ * @author Jose Jimenez-Berni <berni@ias.csic.es>
+ *
+ */
+
+#pragma once
+
+#include "gps_helper.h"
+#include "../../definitions.h"
+
+#define NMEA_RECV_BUFFER_SIZE 1024
+#define NMEA_DEFAULT_BAUDRATE 115200
+
+class GPSDriverNMEA : public GPSHelper
+{
+public:
+	/**
+	 * @param heading_offset heading offset in radians [-pi, pi]. It is substracted from the measurement.
+	 */
+	GPSDriverNMEA(GPSCallbackPtr callback, void *callback_user,
+		      sensor_gps_s *gps_position,
+		      satellite_info_s *satellite_info,
+		      float heading_offset = 0.f);
+	virtual ~GPSDriverNMEA() = default;
+
+	int receive(unsigned timeout) override;
+	int configure(unsigned &baudrate, const GPSConfig &config) override;
+
+private:
+	enum class NMEADecodeFlags : int {
+		got_zda = (1 << 0),
+		got_gga = (1 << 1),
+		got_hdt = (1 << 2),
+		got_gns = (1 << 3),
+		got_rmc = (1 << 4),
+		got_gst = (1 << 5),
+		got_gsa = (1 << 6),
+		got_gsv = (1 << 7),
+		got_vtg = (1 << 8)
+	};
+
+	enum class NMEADecodeState {
+		uninit,
+		got_sync1,
+		got_asteriks,
+		got_first_cs_byte,
+	};
+
+	void decodeInit(void);
+	int handleMessage(int len);
+	int parseChar(uint8_t b);
+
+	int32_t read_int();
+	double read_float();
+	char read_char();
+
+	sensor_gps_s *_gps_position {nullptr};
+	satellite_info_s *_satellite_info {nullptr};
+	uint32_t _last_POS_timeUTC{0};
+	uint32_t _last_VEL_timeUTC{0};
+	uint32_t _last_FIX_timeUTC{0};
+	uint64_t _last_timestamp_time{0};
+
+	uint8_t sat_num_gga{0};
+	uint8_t sat_num_gns{0};
+	uint8_t sat_num_gsv{0};
+	uint8_t sat_num_gpgsv{0};
+	uint8_t sat_num_glgsv{0};
+	uint8_t sat_num_gagsv{0};
+	uint8_t sat_num_gbgsv{0};
+	uint8_t sat_num_bdgsv{0};
+
+	//nmea protocol decode flags NMEADecodeFlags
+	uint32_t _decode_flags{0};
```

Comment:
Do you intend to use this for something?

Reply:
Not, really, this was also inherited from the original code and now we check the status in a different way. It is ok to remove it.